AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/MutationRetryNotifier.swift

```diff
@@ -0,0 +1,117 @@
+//
+// Copyright 2018-2019 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Foundation
+import AWSPluginsCore
+import Combine
+
+@available(iOS 13.0, *)
+final class MutationRetryNotifier {
+    private var nextSyncTimer: DispatchSourceTimer?
+    private var retryAttemptNumber: Int
+
+    private var handlerQueue: DispatchQueue = DispatchQueue.global(qos: .utility)
+    var retryMutationCallback: () -> Void
+    private var reachabilitySubscription: Subscription?
+
+    init(retryAttemptNumber: Int,
+         advice: RequestRetryAdvice,
+         retryMutationCallback: @escaping () -> Void) {
+        self.retryAttemptNumber = retryAttemptNumber
```

Comment:
We only use `retryAttemptNumber` for logging, and the function in which it's logged is immediately invoked from the initializer, so no need to maintain this as a property. In fact, I'd argue that there's no reason to have it at all in this class.

Reply:
Thanks! I was using it at first to control time-outs, but eventually opted for mocking out the entire retry policy class.